src/Illuminate/Mail/Mailable.php

```diff
@@ -936,6 +973,60 @@ public function attachData($data, $name, array $options = [])
         return $this;
     }
 
+    /**
+     * Embed a file into the message.
+     *
+     * @param  string|\Illuminate\Contracts\Mail\Attachable|\Illuminate\Mail\Attachment  $file
+     * @param  array  $options
+     * @return string
+     */
+    public function embed($file, array $options = [])
+    {
+        if ($file instanceof Attachable) {
+            $file = $file->toMailAttachment();
+        }
+
+        if ($file instanceof Attachment) {
+            return $file->embedIn($this);
+        }
+
+        $options['as'] ??= Str::random();
+
+        $this->embeds = collect($this->embeds)
+            ->push([
+                'file' => $file,
+                'options' => $options,
+            ])
+            ->unique('file')
+            ->all();
+
+        return "cid:{$options['as']}";
+    }
+
+    /**
+     * Embed in-memory data in the message.
+     *
+     * @param  string|\Illuminate\Contracts\Mail\Attachable|\Illuminate\Mail\Attachment  $file
+     * @param  string|null  $name
+     * @param  array  $options
+     * @return string
+     */
+    public function embedData($data, $name = null, array $options = [])
+    {
+        $name ??= Str::random();
+
+        $this->rawEmbeds = collect($this->rawEmbeds)
+            ->push([
+                'data' => $data,
+                'name' => $name,
+                'options' => $options,
+            ])
+            ->unique(fn ($file) => $file['name'].$file['data'])
+            ->all();
+
+        return "cid:{$name}";
+    }
+
```

Comment:
I've given these functions the same API as the `attach` function for consistency, however `$name` is option in the data function.

This does mean that it is inconsistent with the `Mail\Message` embed API, but I think that is okay as that class is kind mostly internal and I think it should be made consistent with these APIs in a future major version.